plugins/woocommerce-blocks/assets/js/base/components/notice-banner/style.scss

```diff
@@ -150,3 +150,8 @@
 		@extend %info;
 	}
 }
+
+// To fix notice banner width for all themes.
+.woocommerce.wc-block-store-notices.alignwide {
+	max-width: var(--wp--style--global--wide-size);
+}
```

Comment:
This is to fix the notice banner width for twenty-twenty-four and all other themes. 

| Before | After |
|---|---|
|![image](https://github.com/woocommerce/woocommerce/assets/11503784/b695a7e8-3bcb-44c5-b165-f76c98e06419)|![image](https://github.com/woocommerce/woocommerce/assets/11503784/8d132367-1da2-4956-8471-9152c48defd2)|